phpunit.xml.dist

```diff
@@ -31,7 +31,6 @@ It's auto-generated by sonata-project/dev-kit package.
     <php>
         <ini name="precision" value="8" />
         <env name="SYMFONY_DEPRECATIONS_HELPER" value="max[self]=0" />
-        <env name="SYMFONY_PHPUNIT_REMOVE" value="symfony/yaml" />
```

Comment:
Removing this line means the Symfony phpunit bridge will be removing the prophecy package: https://symfony.com/doc/current/components/phpunit_bridge.html#modified-phpunit-script

Why is important to remove the prophecy package? because phpunit bridge will be doing a new folder with a different composer.json and install there all the things, and it wont use `--ignore-platform-reqs`. The thing is that prophecy is not compatible with php 8 YET (https://github.com/phpspec/prophecy/issues/491)

When we remove this line we make sure the modified phpunit version that gets installed with phpunit bridge no longer contains the prophecy package. That's why it is required to add it in our composer.json because we are installing packages with `--ignore-platform-req=php` and that works.

**Good things:**
- We get to test with php 8
- We get to test with phpunit 9.3

**Bad things:**
- We have to modify all composer.json to add prophecy

If we were not using prophecy, that would not have happened, but it will take a lot of time to remove it IMO.
I think we should have a php8 build running, even if it is failing (but failing executing tests, not on previous steps)

We have an upgrade path for php 8.0:

- Apply this to dev-kit
- Fix all repositories adding prophecy as a dependency (those needing it)
- Take a look at failing test and fix them
- Allow php 8 on repositories without failing tests
- Once all repositories works, add 8.0 as a main supported version and remove `allowed_failures`

And I think we have an upgrade path when php 7.2 support ends in 3 months (https://www.php.net/supported-versions.php):

- Drop php 7.2 from all packages
- Use phpunit 9.3 for all builds
- Add https://github.com/phpspec/prophecy-phpunit to avoid deprecation warnings from phpunit (the integration will be removed on phpunit 10)
- Use rector to upgrade our codebase: https://github.com/rectorphp/rector/blob/master/docs/rector_rules_overview.md#addprophecytraitrector

Or we reconsider supporting prophecy and we start the slow migration to not using it, so one day we can remove the prophecy package and this line, and be happy with it. Both ways are good, but we should really start testing with php 8.0

WDYT @sonata-project/contributors ?

Reply:
Cf https://github.com/sonata-project/dev-kit/issues/624. I would go with the prophecy removal.

Reply:
But what we do in the meantime ? no php 8 support because we can't see what is failing?

Reply:
Solved, now php8 will be able to install and we will not need to add anything to this composer.json